docs/architecture/grpc-for-wcf-developers/ws-protocols.md

```diff
@@ -0,0 +1,31 @@
+---
+title: WS-* protocols - gRPC for WCF Developers
+description: Review of the WS-* protocols supported by WCF and alternatives available with gRPC
+author: markrendle
+ms.date: 09/02/2019
+---
+
+# WS-\* protocols
+
+One of the real benefits of working with Windows Communication Foundation (WCF) was that it supported many of the existing _WS-\*_ standard protocols. This section will briefly cover how gRPC manages the same WS-\* protocols and discuss what options are available when there's no alternative.
+
+## Metadata Exchange - WS-Policy, WS-Discovery, and so on
+
+SOAP services expose Web Services Description Language (WSDL) schema documents with information such as data formats, operations, or communication options. This schema could be used to generate the client code.
```

Comment:
WS-Discovery is used for more than a dynamic client. For example, although it doesn't use WS-Discovery, when you first run outlook, you provide your email address and it uses the domain part to do autodiscovery (using dns) to find your exchange server. So you don't need to use server reflection, you could have a client created from a known .proto but need to discover where your server which implements that .proto can be found. DNS based auto-discovery might be an alternative (which is what outlook/exchange actually uses).